packages/tools/src/utilities/contourSegmentation/copyAnnotation.ts

```diff
@@ -0,0 +1,149 @@
+    );
+  }
+
+  return newAnnotation;
+}
+
+function getAnnotationsMap(segmentationId: string) {
+  const segmentation = getSegmentation(segmentationId);
+  if (!segmentation) {
+    return;
+  }
+
+  const contourRepresentationData = segmentation.representationData
+    ?.Contour as ContourSegmentationData;
+
+  if (!contourRepresentationData) {
+    return;
+  }
+
+  const { annotationUIDsMap } = contourRepresentationData;
+  if (!annotationUIDsMap) {
+    return;
+  }
+  return annotationUIDsMap;
+}
+
+export function copyContourSegment(
+  segmentationId: string,
+  segmentIndex: number,
+  targetSegmentationId: string,
+  targetSegmentIndex: number
+) {
+  const annotationUIDsMap = getAnnotationsMap(segmentationId);
+  const targetAnnotationUIDsMap = getAnnotationsMap(targetSegmentationId);
+  if (!annotationUIDsMap || !targetAnnotationUIDsMap) {
+    return;
+  }
+
+  if (!annotationUIDsMap?.has(segmentIndex)) {
+    return;
+  }
+
+  const annotationUIDs = annotationUIDsMap.get(segmentIndex);
+
+  const viewport = getViewportAssociatedToSegmentation(targetSegmentationId);
+  if (!viewport) {
+    return;
+  }
+  const toolGroup = getToolGroupForViewport(viewport.id);
+
+  const copyContourAnnotation = (annotation: ContourSegmentationAnnotation) => {
+    const newAnnotation = copyAnnotation(
+      annotation,
+      targetSegmentationId,
+      targetSegmentIndex
+    );
+    if (toolGroup) {
+      const instance = toolGroup.getToolInstance(annotation.metadata.toolName);
+      if (instance) {
+        // Properly test if the function isSplineAnnotation exists in instance
+        if (
+          typeof instance.isSplineAnnotation === 'function' &&
+          instance.isSplineAnnotation(annotation)
+        ) {
+          instance.createSplineObjectFromType(
+            newAnnotation,
+            (annotation.data.spline as { type: string }).type
+          );
+        }
+      }
+    }
+    addAnnotation(newAnnotation, viewport.element);
+    newAnnotationsUID.add(newAnnotation.annotationUID);
+    return newAnnotation;
+  };
+
+  const newAnnotationsUID = new Set<string>();
+  for (const annotationUID of annotationUIDs) {
+    const annotation = getAnnotation(
+      annotationUID
+    ) as ContourSegmentationAnnotation;
+    const newAnnotation = copyContourAnnotation(annotation);
+    if (annotation?.childAnnotationUIDs) {
+      newAnnotation.childAnnotationUIDs = [];
+      for (const childAnnotationUID of annotation.childAnnotationUIDs) {
+        const childAnnotation = getAnnotation(
+          childAnnotationUID
+        ) as ContourSegmentationAnnotation;
+        const newChildAnnotation = copyContourAnnotation(childAnnotation);
+        newChildAnnotation.parentAnnotationUID = newAnnotation.annotationUID;
+        newAnnotation.childAnnotationUIDs.push(
+          newChildAnnotation.annotationUID
+        );
+      }
+    }
+  }
+
+  targetAnnotationUIDsMap.set(targetSegmentIndex, newAnnotationsUID);
+}
```

Comment:
we have `getAnnotationMapFromSegmentation` in `packages/tools/src/stateManagement/segmentation/utilities/getAnnotationMapFromSegmentation.ts` that basically does the same thing, if not modify it so that it does it